### ISSUE TITLE
Use https to download root.json file from repositories by default

### ISSUE THREAD

**Issue by poscat0x04** [haskell/cabal#8474]
**Describe the bug**
Currently cabal defaults to use http to download root.json file

**To Reproduce**

1. remove `~/.cabal`
2. `cabal init` a repository
3. `cabal build --verbose | grep curl`

**Expected behavior**
cabal should use https *by default*. It is quite jarring that cabal chooses http as the default protocol to download `root.json` (and other stuff) instead of https in this day and age.

**System information**
``` 
cabal-install version 3.8.1.0
compiled using version 3.8.1.0 of the Cabal library
```

**Comment by Mikolaj** [comment]
Any downsides to using https without a fallback to http? E.g., I don't know, is there a scenario where somebody's firewall would block https, but http?

**Comment by gbaz** [comment]
I do not think that using https rather than http is at all a security improvement. The root.json is itself signed, and that's what provides the security, not the transport layer.

**Comment by poscat0x04** [comment]
@gbaz How does cabal acquire the public keys of the private keys used to generate the signatures of root.json then?

**Comment by gbaz** [comment]
The initial set of root keys is baked into the cabal binary. Those can also sign others in turn: https://hackage.haskell.org/package/hackage-security

https://www.well-typed.com/blog/2015/07/hackage-security-alpha/

**Comment by Mikolaj** [comment]
@poscat0x04: does this answer your question?

**Comment by poscat0x04** [comment]
Yes. This definitely isn't as much of a deal as I initially thought. Still though, I think it would be a good idea to use https by default.
